### PR TITLE
Remove Election and create Membership trait

### PR DESCRIPTION
**Pull request by bfish713**
This creates the `Membership` trait which should contain all the logic related to membership in a comittee or stake table and validating other's membership.

Also removes `Election` trait.  Any thing in election that didn't end up in `Membership` are moved into ConsensusAPI and should be moved `ConsensusExchange` once it exists

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -127,7 +129,7 @@ pub struct HotShotInner<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     storage: I::Storage,
 
     /// This `HotShot` instance's election backend
-    election: I::Election,
+    election: I::Membership,
```
Comment (shenkeyao) [inline src/lib.rs:132]: Nit: rename this field to `membership`.

File: src/traits/networking/libp2p_network.rs [hunk 2]
```diff
@@ -114,7 +114,7 @@ impl<
         TYPES: NodeType,
         PROPOSAL: ProposalType<NodeType = TYPES>,
         VOTE: VoteType<TYPES>,
-        ELECTION: Election<TYPES>,
+        ELECTION: Membership<TYPES>,
```
Comment (shenkeyao) [inline src/traits/networking/libp2p_network.rs:117]: Nit: Can we change `ELECTION` to `MEMBERSHIP` for `TestableNetworkingImplementation`, `Libp2pCommChannel`, and perhaps some other similar traits/structs?

It's probably easier to just replace all `ELECTION` and `election` in the codebase.

File: testing/tests/centralized_server.rs [hunk 3]
```diff
@@ -43,31 +43,11 @@ async fn centralized_server_network_vrf() {
...
-                VrfImpl<
-                    VrfTestTypes,
-                    ValidatingLeaf<VrfTestTypes>,
-                    BLSSignatureScheme<Param381>,
-                    BLSVRFScheme<Param381>,
-                    Hasher,
-                    Param381,
-                >,
-            >,
+            ValidatingProposal<VrfTestTypes, ValidatingLeaf<VrfTestTypes>>,
```
Comment (shenkeyao) [inline testing/tests/centralized_server.rs:46]: Nice!

**Review by shenkeyao — DISMISSED** [review]
Only nit comments. Looks good!

**Review by DieracDelta — APPROVED** [review]
LGTM! The simplified generics make things much nicer. 🎉

**Comment by bfish713** [comment]
Sorry, I didn't realize I needed to handle one more set of merge conflicts, so now I need another approval

**Review by DieracDelta — APPROVED** [review]

**Review by shenkeyao — APPROVED** [review]
LGTM!

**Pull request merged**